app/views/reports/regions/_hypertension_patient_coverage.html.erb

```diff
@@ -26,32 +26,38 @@
     ></div>
     <div class="flex-1 br-tr-4px br-br-4px bg-grey-light"></div>
   </div>
-<% elsif !(@region.estimated_population&.show_coverage) && !(accessible_region?(@region, :manage)) %>
-  <div class="mb-16px p-12px br-4px bg-yellow-light">
+<% elsif !(@region.estimated_population&.show_coverage) %>
     <% if @region.district_region? %>
-      <p class="m-0px fs-14px c-black">
-        Ask a Dashboard admin to add the est. hypertensive population for <%= @region.name %> to see the hypertension coverage rate.
-      </p>
+      <% if (accessible_region?(@region, :manage)) %>
+        <div class="mb-16px p-12px br-4px bg-blue-light">
+          <p class="m-0px fs-14px c-black">
+            Add the est. hypertensive population to see the coverage rate.
+          </p>
+          <%= link_to "+ Add population", edit_admin_facility_group_path(@region.source), :class => "fs-14px" %>
+        </div>
+      <% else %>
+        <div class="mb-16px p-12px br-4px bg-yellow-light">
+          <p class="m-0px fs-14px c-black">
+            Ask a Dashboard admin to add the est. hypertensive population for <%= @region.name %> to see the hypertension coverage rate.
+          </p>
+        </div>
+      <% end %>
     <% elsif @region.state_region? %>
-      <p class="m-0px fs-14px c-black">
-        Ask a Dashboard admin to add the est. hypertensive population for all districts ine<%= @region.name %> to see the hypertension coverage rate.
-      </p>
+      <% if @region.district_regions.all? { |district_region| accessible_region?(district_region, :manage) } %>
```

Comment:
This could be really slow for large states as it will do n+1 calls for every district.  Can we do this in a more efficient way?

Reply:
Switched to a `superset?` check, so we'll fetch the `accessible_district_regions` up front.